.github/workflows/manual-deploy-dev-staging.yml

```
@@ -18,11 +18,98 @@ on:
 env:
   DEVOPS_CHANNEL_ID: C37M86Y8G #devops-deploys
   VETS_WEBSITE_CHANNEL_ID: C02V265VCGH # status-vets-website
+  BUILD_ENV_DEV: vagovdev
+  BUILD_ENV_STAGING: vagovstaging
 
 jobs:
+  build:
+    name: Build
+    runs-on: self-hosted
+
+    env:
+      NODE_EXTRA_CA_CERTS: /etc/ssl/certs/VA-Internal-S2-RCA1-v1.cer.pem
+      
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.commit_sha }}
+
+      - name: Install dependencies
+        uses: ./.github/workflows/install
+        with:
+          key: ${{ hashFiles('yarn.lock') }}
+          yarn_cache_folder: .cache/yarn
+          path: |
+            .cache/yarn
+            node_modules
+
+      - name: Build dev
+        if: ${{ github.event.inputs.deploy_environment == 'dev' || github.event.inputs.deploy_environment == 'both' }}
+        run: yarn build --verbose --buildtype=${{ env.BUILD_ENV_DEV }}
+        timeout-minutes: 30  
+        
+      - name: Generate build details for dev
+        if: ${{ github.event.inputs.deploy_environment == 'dev' || github.event.inputs.deploy_environment == 'both' }}
+        run: |
+          cat > build/${{ env.BUILD_ENV_DEV }}/BUILD.txt << EOF
+          BUILDTYPE=${{ env.BUILD_ENV_DEV }}
+          NODE_ENV=production
+          BRANCH_NAME=$(echo "${GITHUB_REF#refs/heads/}")
+          CHANGE_TARGET=null
+          RUN_ID=${{ github.run_id }}
+          RUN_NUMBER=${{ github.run_number }}
+          REF=${{ github.sha }}
```

So that we get the right sha in the `BUILD.txt`
```suggestion
          REF=${{ github.event.inputs.commit_sha }}
```